File: dotnet/aws/Mcma.Aws.LambdaWorkerInvoker/LambdaWorkerInvoker.cs

using System.Threading.Tasks;
using Amazon.Lambda;
using Amazon.Lambda.Model;
using Mcma.Api;
using Mcma.Context;
using Mcma.Serialization;

namespace Mcma.Aws.Lambda
{
    public class LambdaWorkerInvoker : WorkerInvoker
    {
        public LambdaWorkerInvoker(IContextVariableProvider contextVariableProvider)
            : base(contextVariableProvider)
        {
        }

        protected override async Task InvokeAsync(string workerFunctionId, WorkerRequest request)
        {
            // invoking worker lambda function that will handle the work for the service
            using (var lambdaClient = new AmazonLambdaClient())
                await lambdaClient.InvokeAsync(
                    new InvokeRequest
                    {
                        FunctionName = workerFunctionId,
                        InvocationType = "Event",
                        LogType = "None",
                        Payload = request.ToMcmaJson().ToString()
                    }
                );
        }
    }
}

File: nodejs/aws/lambda-worker-invoker/lib/lambda-worker-invoker.ts
import { Lambda } from "aws-sdk";
import { InvokeWorker, WorkerInvoker, WorkerRequest } from "@mcma/api";

const lambdaClient = new Lambda({ apiVersion: "2015-03-31" });

export const invokeLambdaWorker: InvokeWorker = async (workerFunctionId: string, workerRequest: WorkerRequest): Promise<void> => {
    await lambdaClient.invoke({
        FunctionName: workerFunctionId,
        InvocationType: "Event",
        LogType: "None",
        Payload: JSON.stringify(workerRequest)
    }).promise();
};

export class LambdaWorkerInvoker extends WorkerInvoker {
    constructor() {
        super(invokeLambdaWorker);
    }
}
